rekall-core/rekall/plugins/linux/heap_analysis.py

```diff
@@ -0,0 +1,4388 @@
+        thread_stack_offsets = []
+        thread_group_offset = self.profile.get_obj_offset("task_struct", "thread_group")
+        for thread_group in task.thread_group.walk_list("next"):
+            thread_task = self.profile.Object("task_struct",
+                                              offset=thread_group.obj_offset - thread_group_offset,
+                                              vm=self.process_as)
+
+            pt_regs = self.profile.Object(
+                "pt_regs",
+                offset=thread_task.thread.sp0 - self.profile.get_obj_size("pt_regs"),
+                vm=self.process_as)
+
+            temp = dict()
+            temp['ebp'] = pt_regs.bp.v()
+            temp['esp'] = pt_regs.sp.v()
+            temp['pid'] = thread_task.pid.v()
+            thread_stack_offsets.append(temp)
+
+
+        # The first pair contains the "main" thread and the mm start_stack value is more reliable
+        # for identifying the relevant memory region than the saved frame pointers
+        thread_stack_offsets[0]['start_stack'] = task.mm.start_stack
+
+        heap_area = False
+        for vma in task.mm.mmap.walk_list("vm_next"):
+            temp_vma = dict()
+
+            if vma.vm_file:
+                fname = task.get_path(vma.vm_file)
+                if heap_area:
+                    heap_area = False
+
+            else:
+                fname = ""
+                if heap_area:
+                    fname = self._heap_vma_identifier
+
+                else:
+                    fname = self._pot_mmapped_vma_identifier
+
+
+                # main heap can have 3 or more vm_area_struct structs
+                if vma.vm_start <= task.mm.start_brk <= vma.vm_end or \
+                        task.mm.start_brk <= vma.vm_start < vma.vm_end <= task.mm.brk or \
+                        vma.vm_start <= task.mm.brk <= vma.vm_end:
+                    fname = self._main_heap_identifier
+                    heap_area = True
+
+                else:
+                    for offsets in thread_stack_offsets:
+                        if ('start_stack' in offsets.keys()
+                                and vma.vm_start <= offsets['start_stack'] <= vma.vm_end) \
+                                or vma.vm_start <= offsets['ebp'] <= vma.vm_end \
+                                or vma.vm_start <= offsets['esp'] <= vma.vm_end:
+                            fname = "[stack"
+                            pid = offsets['pid']
+                            fname += "]" if task.pid == pid else ":{:d}]".format(pid)
+
+                            temp_vma['ebp'] = offsets['ebp']
+                            temp_vma['esp'] = offsets['esp']
+
+                            heap_area = False
+
+            temp_vma['name'] = fname
+            temp_vma['vma'] = vma
+            result.append(temp_vma)
+
+        return sorted(result, key=lambda vma: vma['vma'].vm_start)
+
+
+    def _load_libc_profile(self):
```

Comment:
I think this function should not be needed. It should be possible to just get the glibc profile from the address resolver. Doing this means that people can use specialized profiles in a standard way (just like any other dll/module).